crates/virtio-queue/src/lib.rs

```diff
@@ -574,7 +575,8 @@ impl<M: GuestAddressSpace> QueueState<M> {
     // Helper method that writes `val` to the `avail_event` field of the used ring, using
     // the provided ordering.
     fn set_avail_event(&self, mem: &M::T, val: u16, order: Ordering) -> Result<(), Error> {
-        let offset = (4 + self.actual_size() * 8) as u64;
+        let elem_sz = VIRTQ_USED_ELEMENT_SIZE * u64::from(self.actual_size());
```

Comment:
Can you help with fixing these ones as well: https://github.com/rust-vmm/vm-virtio/blob/main/crates/virtio-queue/src/lib.rs#L626, https://github.com/rust-vmm/vm-virtio/blob/main/crates/virtio-queue/src/lib.rs#L790 so we can close the issue?
I think I would also define constants for [the alignment](https://github.com/rust-vmm/vm-virtio/blob/main/crates/virtio-queue/src/lib.rs#L696-L702) of the queue parts, e.g. `pub(crate) const DESC_TABLE_ALIGNMENT: u64 = 0Xf;`, but we can leave it for later as well.
Thanks!

Reply:
Sure. I did a cursory search for offset so missed those.